### DISCUSSION TITLE
posts/Go/vscode/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#368] [Announcements]
https://www.liwenzhou.com/posts/Go/vscode/

 李文周的Blog

**Comment by RiversLee** [comment]
您好，mac的git tools可以在linux上面使用吗

**Comment by Q1mi** [comment]
> @RiversLee
> 您好，mac的git tools可以在linux上面使用吗

这个我没有试过呢。

**Comment by cg430** [comment]
打卡

**Comment by Brand123147** [comment]
> @RiversLee
> 您好，mac的git tools可以在linux上面使用吗
亲测ubuntu可以用啊

**Comment by Brand123147** [comment]
ubuntu装好麻烦，看到网上有些直接一个一个装的太辛苦了，还好找到了你归纳好的谢啦，放到bin下改一下可执行权限perfect

**Comment by lzb-by** [comment]
你好，插件安装完了，提示过期，更新又更新不了。怎么搞？

**Comment by Brand123147** [comment]
> @lzb-by
> 你好，插件安装完了，提示过期，更新又更新不了。怎么搞？

我是重新安装，看这个的：https://blog.csdn.net/qq_15889613/article/details/103378602
还有什么骚操作欢迎告知啊

**Comment by shuitianyise** [comment]
博主 你好我是在windows上安装了 vscode  我想远程那个  centos上的 go开发环境， centos上 的$GOROOT $GOPATH 我都设置好了 但是在vscode上我不知道 怎么设置？  你那里 有尝试过这样的问题吗？

**Comment by shuitianyise** [comment]
博主 我感觉 你这个
“第四步：执行git clone https://github.com/golang/lint.git命令”  写错了
是不是 应该这样啊？“第四步：执行git clone https://github.com/golang/lint.git lint命令”

**Comment by subB1ss** [comment]
我测试了下在win平台上配置GOPROXY可以解决VSCODE的大部分go tools安装失败问题。剩下的几个可以通过自己手动安装解决。顺便厚颜无耻的想要伸手问一下老师教学视频的VSCODE主题和字体是什么。

**Comment by limin199510** [comment]
不同版本的go和vscode可以直接用百度云中的go-tools吗

**Comment by tuohuang** [comment]
按照方法一 clone完后 在vscode执行Go:Install/Update Toolsa还是全部失败

**Comment by tuohuang** [comment]
WSL 下按照方法一 clone完后 在vscode执行Go:Install/Update Tools还是全部失败

**Comment by tuohuang** [comment]
应该是在GOROOT目录下进行操作

**Comment by Q1mi** [comment]
> @limin199510
> 不同版本的go和vscode可以直接用百度云中的go-tools吗

不同版本是不通用的哦

**Comment by Terryjans** [comment]
你好，我是哔哩哔哩来的，视频里有百度云的安装包，这个页面怎么没有了

**Comment by Q1mi** [comment]
> @Terryjans
> 你好，我是哔哩哔哩来的，视频里有百度云的安装包，这个页面怎么没有了

博客更新了一下，按最新博客上的方法就行，
互联网技术更新迭代太快了，没办法一直录新视频。

**Comment by 13485507836** [comment]
安装tool还是失败

**Comment by dema-go** [comment]
按照博客内容下载tools还是显示失败。。。怎么办？？

**Comment by Brand123147** [comment]
> @IdlwlmaU
> 按照博客内容下载tools还是显示失败。。。怎么办？？

要不试试重启计算机？我有过一次搞了很久就是下载失败，第二天打开电脑啥都没操作，直接下载就成功了。。。后来百度到说配置完成要关掉IDE，或者重启电脑再来下载

**Comment by Q1mi** [comment]
> @IdlwlmaU
> 按照博客内容下载tools还是显示失败。。。怎么办？？

有报错提示么？

**Comment by dema-go** [comment]
> @Q1mi
> > @IdlwlmaU
> > 按照博客内容下载tools还是显示失败。。。怎么办？？
> 
> 有报错提示么？
已经解决了，谢谢啦

**Comment by dema-go** [comment]
写go程序的时候出现错误但是没有红色的波形曲线。怎么才能让这个错误提示显示出来啊？？看了很多博客也没解决。

**Comment by cn1001wang** [comment]
设置了go env -w GOPROXY =...之后就可以了。我这里安装成功

**Comment by qq1430516711** [comment]
为什么我的是安装失败。是哪一步错了？

**Comment by soldierbupt** [comment]
怎么实现鼠标指示某个语句，自动提示补全缺少的包

**Comment by mxbmxl** [comment]
怎样查看“类图”：结构体依赖关系图、结构体实现了哪些接口？

**Comment by mxbmxl** [comment]
VSCode开发Go，怎样查看“类图”：比如结构体依赖关系图、结构体实现了哪些接口？

**Comment by lzl0601** [comment]
李老师，你的视频里面有讲到这个博客里面有go-tool的云盘链接，我没找到啊

**Comment by Q1mi** [comment]
> @lzl0601
> 李老师，你的视频里面有讲到这个博客里面有go-tool的云盘链接，我没找到啊

现在都用gomodule了，可以设置GOPROXY解决。go env -w GOPROXY=https://goproxy.cn,direct

**Comment by SkyTine** [comment]
6啊，不仅仅是单纯的配置，一套流畅下来会少踩很多坑

**Comment by jiaxin0521** [comment]
9 tools failed to install.

gopkgs: failed to install gopkgs(github.com/uudashr/gopkgs/v2/cmd/gopkgs@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/uudashr/gopkgs/v2/cmd/gopkgs@latest
go install: github.com/uudashr/gopkgs/v2/cmd/gopkgs@latest: module github.com/uudashr/gopkgs/v2/cmd/gopkgs: Get "https://proxy.golang.org/github.com/uudashr/gopkgs/v2/cmd/gopkgs/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
go-outline: failed to install go-outline(github.com/ramya-rao-a/go-outline@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/ramya-rao-a/go-outline@latest
go install: github.com/ramya-rao-a/go-outline@latest: module github.com/ramya-rao-a/go-outline: Get "https://proxy.golang.org/github.com/ramya-rao-a/go-outline/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
gotests: failed to install gotests(github.com/cweill/gotests/gotests@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/cweill/gotests/gotests@latest
go install: github.com/cweill/gotests/gotests@latest: module github.com/cweill/gotests/gotests: Get "https://proxy.golang.org/github.com/cweill/gotests/gotests/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
gomodifytags: failed to install gomodifytags(github.com/fatih/gomodifytags@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/fatih/gomodifytags@latest
go install: github.com/fatih/gomodifytags@latest: module github.com/fatih/gomodifytags: Get "https://proxy.golang.org/github.com/fatih/gomodifytags/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
impl: failed to install impl(github.com/josharian/impl@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/josharian/impl@latest
go install: github.com/josharian/impl@latest: module github.com/josharian/impl: Get "https://proxy.golang.org/github.com/josharian/impl/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
goplay: failed to install goplay(github.com/haya14busa/goplay/cmd/goplay@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/haya14busa/goplay/cmd/goplay@latest
go install: github.com/haya14busa/goplay/cmd/goplay@latest: module github.com/haya14busa/goplay/cmd/goplay: Get "https://proxy.golang.org/github.com/haya14busa/goplay/cmd/goplay/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
dlv: failed to install dlv(github.com/go-delve/delve/cmd/dlv@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v github.com/go-delve/delve/cmd/dlv@latest
go install: github.com/go-delve/delve/cmd/dlv@latest: module github.com/go-delve/delve/cmd/dlv: Get "https://proxy.golang.org/github.com/go-delve/delve/cmd/dlv/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
staticcheck: failed to install staticcheck(honnef.co/go/tools/cmd/staticcheck@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v honnef.co/go/tools/cmd/staticcheck@latest
go install: honnef.co/go/tools/cmd/staticcheck@latest: module honnef.co/go/tools/cmd/staticcheck: Get "https://proxy.golang.org/honnef.co/go/tools/cmd/staticcheck/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
 
gopls: failed to install gopls(golang.org/x/tools/gopls@latest): Error: Command failed: C:\Program Files\Go\bin\go.exe install -v golang.org/x/tools/gopls@latest
go install: golang.org/x/tools/gopls@latest: module golang.org/x/tools/gopls: Get "https://proxy.golang.org/golang.org/x/tools/gopls/@v/list": dial tcp 142.251.42.241:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

**Comment by jiaxin0521** [comment]
请问老师我九个工具都下载不了该怎么办

**Comment by Syoehn** [comment]
打卡

**Comment by GYPgalaxy** [comment]
如果有windows的兄弟配置tools时候failed，可以试试以下方法：

cmd输入go env -w GOPROXY=https://proxy.golang.com.cn,direct 并回车
cmd输入go clean --modcache并回车
重启vscode再进行后续下载tools操作
